tests/aws/services/lambda_/test_lambda_api.py

```diff
@@ -2626,6 +2626,16 @@ def test_lambda_provisioned_lifecycle(self, create_lambda_function, snapshot, aw
             )
         snapshot.match("put_provisioned_on_alias_versionconflict", e.value.response)
 
+        def _wait_provisioned():
+            status = aws_client.lambda_.get_provisioned_concurrency_config(
+                FunctionName=function_name, Qualifier=function_version
+            )["Status"]
+            if status == "FAILED":
+                raise ShortCircuitWaitException("terminal fail state")
```

Comment:
nice pattern we should remember and use more often in tests 👍

Reply:
FYI: also a better solution than the regular retry from the Lambda invocation loop 👍 (favored this one in rebase)